content-src/components/TopSites/TopSiteFormInput.jsx

```diff
@@ -60,22 +59,22 @@ export class TopSiteFormInput extends React.PureComponent {
     const {typeUrl} = this.props;
     const {validationError} = this.state;
 
-    return (<label><FormattedMessage id={this.props.titleId} />
+    return (<label>
+      <span data-l10n-id={this.props.titleId} />
```

Comment:
Changing here to <label data-l10n-id={this.props.titleId}> breaks the layout. @Mardak feel free to chime if there is a way to  remove span tag here.

Reply:
The way this is done is setting a `data-l10n-name` on the *inline* children and putting empty nodes in the fluent string:
https://github.com/mozilla/activity-stream/blob/8b0e7e99f392bf4bbbfe2fe03878ef205befb050/content-src/asrouter/templates/Trailhead/Trailhead.jsx#L280-L284
along with
```ini
onboarding-join-form-legal = By proceeding, you agree to the <a data-l10n-name="terms">Terms of Service</a> and <a data-l10n-name="privacy">Privacy Notice</a>.
```

So in this case,
```ini
newtab-topsites-title-label = Title <div data-l10n-name="input"></div>
```
```jsx
<label>
  <div data-l10n-name="input" className=…>
    <input … />
  </div>
</label>
```
 
However, the child here is actually a block and the text spacing doesn't actually matter, so we don't actually want to do the above. 😉 So we're good with the extra span in this case. I suppose we could make the `<label for=…>` have no children, but then we would need to reference the `<input>` by id.

Some more details here https://firefox-source-docs.mozilla.org/intl/l10n/l10n/fluent_migrations.html#concatenating-strings